src/draw/nxp/vglite/lv_draw_vglite.c

```diff
@@ -311,9 +334,22 @@ static int32_t _vglite_dispatch(lv_draw_unit_t * draw_unit, lv_layer_t * layer)
     return 1;
 }
 
+#if LV_USE_VGLITE_DRAW_ASYNC
+static int32_t _vglite_wait_for_finish(lv_draw_unit_t * draw_unit)
+{
+    lv_draw_vglite_unit_t * draw_vglite_unit = (lv_draw_vglite_unit_t *) draw_unit;
+    draw_vglite_unit->wait_for_finish = true;
+
+    if(draw_vglite_unit->inited)
+        lv_thread_sync_signal(&draw_vglite_unit->sync);
```

Comment:
Should this be a wait?

Reply:
No, it has to be sync.

This API gets called by the LVGL thread and will signal the VGLite thread to run and complete all pending tasks.

Reply:
I think I understand your concern...

The LVGL main thread calls `lv_draw_wait_for_finish();`
This means that the draw unit gets signaled to wait for completing the pending queue. The draw unit thread will actually wait for completion and signal back the tasks ready status.